cedar-policy-core/src/parser/err.rs

```diff
@@ -822,27 +844,18 @@ impl<T: Into<ParseError>> Extend<T> for ParseErrors {
 
 impl IntoIterator for ParseErrors {
     type Item = ParseError;
-    type IntoIter = std::vec::IntoIter<Self::Item>;
+    type IntoIter = iter::Chain<iter::Once<Self::Item>, std::vec::IntoIter<Self::Item>>;
 
     fn into_iter(self) -> Self::IntoIter {
-        self.0.into_iter()
+        iter::once(self.0.head).chain(self.0.tail)
```

Comment:
One more comment: `NonEmpty` does implement `IntoIterator` itself, so this can be just
```suggestion
        self.0.into_iter()
```
and the `type IntoIter` would be `<NonEmpty<ParseError> as IntoIterator>::IntoIter`